stories/core/Typography/Label.stories.tsx

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { Story, Meta } from '@storybook/react';
+import React from 'react';
+import { Input, Label, LabelProps, Text } from '../../../src/core';
+
+export default {
+  component: Label,
+  args: {
+    children: 'Name',
+    displayStyle: 'block',
+    required: false,
+    as: 'label',
+  },
+  title: 'Typography/Label',
+  decorators: [
+    (Story) => (
+      <div style={{ maxWidth: 'clamp(300px, 50%, 100%)' }}>
+        <Story />
+      </div>
+    ),
+  ],
+} as Meta<LabelProps>;
+
+export const Basic: Story<LabelProps> = (args) => {
+  return (
+    <>
+      <Label htmlFor='text-input' {...args}>
+        Name
+      </Label>
+      <Input id='text-input' placeholder='Enter name' />
+    </>
+  );
+};
+Basic.args = {};
+
+export const Inline: Story<LabelProps> = (args) => {
+  return (
+    <div style={{ display: 'flex' }}>
+      <Label htmlFor='text-input' displayStyle='inline' required {...args}>
+        Name
+      </Label>
+      <Input id='text-input' defaultValue='James Bond' required />
+    </div>
+  );
+};
+Inline.args = {
+  displayStyle: 'inline',
+  required: true,
+};
+
+export const Polymorphic: Story<LabelProps<'div'>> = (args) => {
+  return (
+    <div style={{ display: 'flex' }}>
+      <Label displayStyle='inline' as='div' {...args}>
+        <Text isMuted>Name:</Text>
```

Comment:
I was not able to use `<Label as={Text} isMuted>`. I think there is a problem in the polymorphic approach where it doesn't take the props of custom component

Reply:
I think it is not a big issue right now for `Label`. But you check this article: https://blog.andrewbran.ch/polymorphic-react-components/. It handles custom components and has an alternative way for allowing custom types.

Reply:
I already read that article when I first implemented polymorphism. Their conclusion doesn't really help:
![image](https://user-images.githubusercontent.com/9084735/141997221-0d0230b1-6ba6-498c-af2e-b8b989e2e393.png)

Reply:
Turns out it's because Text is also a polymorphic component, so I just had to add a wrapper like this: `as={(args: TextProps) => <Text {...args} />} isMuted>`
```suggestion
      <Label displayStyle='inline'as={(args: TextProps) => <Text {...args} />} isMuted {...args}>
        Name:
```

But it's not reliable for styling because the components have same specificity (notice the Text font-weight gets overridden by Label)
![image](https://user-images.githubusercontent.com/9084735/142037047-7dadf08c-6f39-42c7-a59a-f22caf93a5bc.png)